docs/examples/tensorflow/tensorflow-dataset-multigpu.ipynb

```diff
@@ -0,0 +1,311 @@
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "WARNING:tensorflow:From <ipython-input-6-fb80f01b9086>:26: flatten (from tensorflow.python.layers.core) is deprecated and will be removed in a future version.\n",
+      "Instructions for updating:\n",
+      "Use keras.layers.flatten instead.\n",
+      "WARNING:tensorflow:From /home/awolant/.local/lib/python3.6/site-packages/tensorflow_core/python/layers/core.py:332: Layer.apply (from tensorflow.python.keras.engine.base_layer) is deprecated and will be removed in a future version.\n",
+      "Instructions for updating:\n",
+      "Please use `layer.__call__` method instead.\n",
+      "WARNING:tensorflow:From <ipython-input-6-fb80f01b9086>:27: dense (from tensorflow.python.layers.core) is deprecated and will be removed in a future version.\n",
+      "Instructions for updating:\n",
+      "Use keras.layers.Dense instead.\n",
+      "WARNING:tensorflow:From /home/awolant/.local/lib/python3.6/site-packages/tensorflow_core/python/ops/resource_variable_ops.py:1630: calling BaseResourceVariable.__init__ (from tensorflow.python.ops.resource_variable_ops) with constraint is deprecated and will be removed in a future version.\n",
+      "Instructions for updating:\n",
+      "If using Keras pass *_constraint arguments to layers.\n",
+      "WARNING:tensorflow:From <ipython-input-6-fb80f01b9086>:28: dropout (from tensorflow.python.layers.core) is deprecated and will be removed in a future version.\n",
+      "Instructions for updating:\n",
+      "Use keras.layers.dropout instead.\n",
+      "WARNING:tensorflow:From <ipython-input-6-fb80f01b9086>:34: softmax_cross_entropy_with_logits (from tensorflow.python.ops.nn_ops) is deprecated and will be removed in a future version.\n",
+      "Instructions for updating:\n",
+      "\n",
+      "Future major versions of TensorFlow will allow gradients to flow\n",
+      "into the labels input on backprop by default.\n",
+      "\n",
+      "See `tf.nn.softmax_cross_entropy_with_logits_v2`.\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "iterator_initializers = []\n",
+    "\n",
+    "with tf.device('/cpu:0'):\n",
+    "    tower_grads = []\n",
+    "\n",
+    "    for i in range(num_gpus):\n",
+    "        with tf.device('/gpu:{}'.format(i)):\n",
+    "            daliset = dali_tf.DALIDataset(\n",
+    "                pipeline=MnistPipeline(BATCH_SIZE, device_id=i, shard_id=i, num_shards=num_gpus),\n",
+    "                batch_size=BATCH_SIZE,\n",
+    "                shapes=shapes,\n",
+    "                dtypes=dtypes,\n",
+    "                device_id=i).with_options(options)\n",
+    "\n",
+    "            iterator = tf.data.make_initializable_iterator(daliset)\n",
+    "            iterator_initializers.append(iterator.initializer)\n",
+    "            images, labels = iterator.get_next()\n",
+    "\n",
+    "            images = tf.reshape(\n",
```

Comment:
Can't we use DALI reshape operator inside the pipeline instead?

Reply:
Actually, we don't need reshaping at all, so I just removed it.